backend/api/Controllers/RobotController.cs

```diff
@@ -214,26 +214,43 @@ public async Task<ActionResult<Robot>> PostRobot([FromBody] Robot robot)
             }
 
             _logger.LogError(e, "Error getting mission from Echo");
-            return new StatusCodeResult(StatusCodes.Status502BadGateway);
+            return StatusCode(StatusCodes.Status502BadGateway, $"{e.Message}");
         }
         catch (JsonException e)
         {
             _logger.LogError(e, "Error deserializing mission from Echo");
-            return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            return StatusCode(StatusCodes.Status500InternalServerError, $"{e.Message}");
```

Comment:
I think in this case (the case of json errors), the exception message should not be passed to the user as it won't help them understand what went wrong and we are not sure what it might contain. We could instead simply pass a message saying that something went wrong during the processing of the response from ISAR.
The same goes for the other endpoints

Reply:
Agreed that the exception message might be generic and not necessarily helpful to users. I reworded the messages relating to `JsonException` for the relevant endpoints.